Sources/CNIOLinux/include/barrier.h

```diff
@@ -0,0 +1,99 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// from liburing project, local copy needed for builds on linux platforms without liburing installed
```

Comment:
two notes here:
- if we vendor any files, we need to add it to `NOTICE.TXT`
- does this file not have an official license header?

Reply:
No, it starts with:

/* SPDX-License-Identifier: MIT */

I can have a look again if I can kill those two files, but IIRC it was a bit messy.